src/sentry/tasks/reprocessing2.py

```diff
@@ -18,10 +18,9 @@
 def reprocess_group(project_id, group_id, offset=0, start_time=None, max_events=None):
     from sentry.reprocessing2 import start_group_reprocessing
 
-    start_group_reprocessing(project_id, group_id, max_events=max_events)
-
     if start_time is None:
         start_time = time.time()
+        start_group_reprocessing(project_id, group_id, max_events=max_events)
```

Comment:
TODO: Write test for this massive bug! Not sure how yet

Reply:
I've reenabled the testsuite in CI because I believe that this bugfix fixes the deadlock.